src/utils/ChatMessageUtils.js

```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+
+export function convertRawMessage(rawMessage, currentThreadID) {
+  return {
+    ...rawMessage,
+    date: new Date(rawMessage.timestamp),
+    isRead: rawMessage.threadID === currentThreadID
+  };
+};
+
+export function getCreatedMessageData(text, currentThreadID) {
+  var timestamp = Date.now();
+  return {
+    id: 'm_' + timestamp,
+    threadID: currentThreadID,
+    authorName: 'You', 
+    date: new Date(timestamp),
+    text: text,
+    isRead: true
+  };
+};
+
+export function getSUSIMessageData(createdMessage, currentThreadID) {
+  var timestamp = Date.now();
+
+  let receivedMessage =  {
+    id: 'm_' + timestamp,
+    threadID: currentThreadID,
+    authorName: 'SUSI', // hard coded for the example
+    date: new Date(timestamp),
+    text: '',
+    isRead: true
+  };
+
+  axios.get('http://api.asksusi.com/susi/chat.json?', {
+    params: {
```

Comment:
Maybe(as I am a novice in react) in this block we can have dataType: jsonp property, and it may resolve the problem of cross origin access.